Frameworks/CoreText/CTParagraphStyle.mm

```diff
@@ -17,235 +17,182 @@
-                    // a default value or it has been set by the user.
-                    // offsetToDefaultStatus is the offset to this struct, which is also in turn the offset to _isDefault bool variable
-                    // within the struct.
-                    size_t offsetToDefaultStatus = propertyInfoTable[settings[i].spec].offset;
-                    auto structContainingData =
-                        reinterpret_cast<void*>((reinterpret_cast<char*>(paraStyleProperty) + offsetToDefaultStatus));
-                    *(reinterpret_cast<bool*>(structContainingData)) = false;
-
-                    if (expectedSize == sizeof(CGFloat)) {
-                        auto property = reinterpret_cast<_CTParagraphStyleFloatProperty*>(structContainingData);
-
-                        size_t offsetForValue = offsetof(_CTParagraphStyleFloatProperty, _value);
-                        auto locationToWrite = reinterpret_cast<char*>(property) + offsetForValue;
-                        *reinterpret_cast<CGFloat*>(locationToWrite) = *reinterpret_cast<const CGFloat*>(settings[i].value);
-
-                    } else if (settings[i].spec == kCTParagraphStyleSpecifierBaseWritingDirection) {
-                        auto property = reinterpret_cast<_CTParagraphStyleCTWritingDirectionProperty*>(structContainingData);
-
-                        size_t offsetForValue = offsetof(_CTParagraphStyleCTWritingDirectionProperty, _value);
-                        auto locationToWrite = reinterpret_cast<char*>(property) + offsetForValue;
-                        *reinterpret_cast<CTWritingDirection*>(locationToWrite) =
-                            *reinterpret_cast<const CTWritingDirection*>(settings[i].value);
-                    } else if (settings[i].spec == kCTParagraphStyleSpecifierAlignment) {
-                        auto property = reinterpret_cast<_CTParagraphStyleCTTextAlignmentProperty*>(structContainingData);
-
-                        size_t offsetForValue = offsetof(_CTParagraphStyleCTTextAlignmentProperty, _value);
-                        auto locationToWrite = reinterpret_cast<char*>(property) + offsetForValue;
-                        *reinterpret_cast<CTTextAlignment*>(locationToWrite) = *reinterpret_cast<const CTTextAlignment*>(settings[i].value);
-
-                    } else if (settings[i].spec == kCTParagraphStyleSpecifierLineBreakMode) {
-                        auto property = reinterpret_cast<_CTParagraphStyleCTLineBreakModeProperty*>(structContainingData);
-
-                        size_t offsetForValue = offsetof(_CTParagraphStyleCTLineBreakModeProperty, _value);
-                        auto locationToWrite = reinterpret_cast<char*>(property) + offsetForValue;
-                        *reinterpret_cast<CTLineBreakMode*>(locationToWrite) = *reinterpret_cast<const CTLineBreakMode*>(settings[i].value);
-                    }
-                }
-            }
+            style->SetValue(settings[i]);
         }
```

Comment:
Update: reference platform goes on its merry way and skips over incorrect values, returning an object with any valid values it was initialized with